src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { UpdatingForm } from '../UpdatingForm';
+
+type Props = {
+  todo: Todo;
+  loadingTodos: number[];
+  deleteTodo: (todoId: number) => void;
+  updateTodo: (
+    todoId: number,
+    newTodoData: Partial<Pick<Todo, 'title' | 'completed'>>
+  ) => void;
+};
+
+export const TodoInfo: React.FC<Props> = ({
+  todo,
+  loadingTodos,
+  deleteTodo,
+  updateTodo,
+}) => {
+  const {
+    id, title, completed,
+  } = todo;
+
+  const [isUpdating, setIsUpdating] = useState(false);
+
+  const handleToggleComplete = async () => {
+    await updateTodo(id, { completed: !todo.completed });
+  };
```

Comment:
Create this function in the app and provide by props instead `updateTodo`

```suggestion
  const handleToggleComplete = () => {
    onUpdateTodo(todo);
  };
```